### PR TITLE
Simulator final work

### PR DESCRIPTION
**Pull request by Nashatyrev**

### CODE REVIEW

File: chain/src/main/java/org/ethereum/beacon/chain/observer/ObservableStateProcessorImpl.java [hunk 1]
```diff
@@ -211,23 +211,62 @@ private synchronized void purgeAttestations(SlotNumber slot) {
...
-      observableStateSink.onNext(new ObservableBeaconState(
-          head.getBlock(), epochState, pendingOperations));
+  private void newSlot(SlotNumber newSlot) {
+    if (head.getBlock().getSlot().greater(newSlot)) {
+      return;
+    }
+    updateCurrentObservableState(head, newSlot);
+  }
+
+  private void updateCurrentObservableState(BeaconTupleDetails head, SlotNumber slot) {
```
Comment (zilm13) [inline chain/src/main/java/org/ethereum/beacon/chain/observer/ObservableStateProcessorImpl.java]: this method looks too complex. a bit of refactoring would make it better!

File: start/config/src/main/java/org/ethereum/beacon/emulator/config/simulator/PeersConfig.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class PeersConfig {
+  private int peersCount = 1;
+  private boolean isValidator = true;
+
+  private long systemTimeShift = 0;
+  private long wireInboundDelay = 0;
```
Comment (zilm13) [inline start/config/src/main/java/org/ethereum/beacon/emulator/config/simulator/PeersConfig.java]: ms?

File: start/config/src/main/java/org/ethereum/beacon/emulator/config/simulator/PeersConfig.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  private long wireOutboundDelay = 0;
+
+  private String blsPrivateKey = null;
```
Comment (zilm13) [inline start/config/src/main/java/org/ethereum/beacon/emulator/config/simulator/PeersConfig.java]: one for all?

File: start/config/src/main/java/org/ethereum/beacon/emulator/config/simulator/PeersConfig.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
+package org.ethereum.beacon.emulator.config.simulator;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class PeersConfig {
+  private int peersCount = 1;
+  private boolean isValidator = true;
+
+  private long systemTimeShift = 0;
```
Comment (zilm13) [inline start/config/src/main/java/org/ethereum/beacon/emulator/config/simulator/PeersConfig.java]: what is the purpose of this?

File: .gitignore [hunk 5]
```diff
@@ -9,3 +9,7 @@ out
 # :pow:ethereumj
 pow/ethereumj/database
 pow/ethereumj/logs
+work
```
[the remark this answers is not in the document]
Comment (Nashatyrev) [reply .gitignore]: This is mine :)
BTW recommended, to set IDEA default working dir to './work' so the logs, db, etc files are not spread across modules subdirs

**Pull request merged**